packages/report/index.js

```diff
@@ -52,47 +56,61 @@ export async function run (options) {
 
   spinner.start('Fetching users page 0...')
 
-  await Promise.all([
-    (async () => {
-      let after = null
-      let i = 1
-      while (true) {
-        const { findUploadsCreatedAfter: uploadsPage } = await retry(() => db.query(FIND_UPLOADS_CREATED_AFTER, { since: from, cursor: after }))
-        totalUploads += uploadsPage.data.length
-
-        for (const upload of uploadsPage.data) {
-          const uploads = userUploads.get(upload.user._id) || []
-          userUploads.set(upload.user._id, uploads.concat(upload))
+  try {
+    await Promise.all([
+      (async () => {
+        let after = null
+        let i = 1
+        while (true) {
+          const { findUploadsCreatedAfter: uploadsPage } = await retry(() => db.query(FIND_UPLOADS_CREATED_AFTER, { since: from, cursor: after }))
+          totalUploads += uploadsPage.data.length
+
+          for (const upload of uploadsPage.data) {
+            const uploads = userUploads.get(upload.user._id) || []
+            userUploads.set(upload.user._id, uploads.concat(upload))
+          }
+
+          spinner.text = renderText(`Fetching uploads page ${i}...`, { totalUsers, totalUploads, users, userUploads }, options)
+          after = uploadsPage.after
+          if (!after) break
+          i++
         }
+      })(),
+      (async () => {
+        let after = null
+        let i = 1
```

Comment:
Why this and iterate 5 times?

Reply:
Ok, you move faster than me https://github.com/web3-storage/web3.storage/pull/324/commits/b01dece4a72807db682893df08cdd276d19d4c46 😂